crates/hyperswitch_domain_models/src/payments/payment_intent.rs

```diff
@@ -287,39 +293,44 @@ pub enum PaymentIntentUpdate {
+        description: Box<Option<common_utils::types::Description>>,
+        return_url: Box<Option<common_utils::types::Url>>,
+        setup_future_usage: Option<storage_enums::FutureUsage>,
+        apply_mit_exemption: Option<common_enums::MitExemptionRequest>,
+        statement_descriptor: Box<Option<StatementDescriptor>>,
+        order_details: Box<Option<Vec<Secret<diesel_models::types::OrderDetailsWithAmount>>>>,
+        allowed_payment_method_types: Box<Option<Vec<common_enums::PaymentMethodType>>>,
+        metadata: Box<Option<pii::SecretSerdeValue>>,
+        connector_metadata: Box<Option<pii::SecretSerdeValue>>,
+        feature_metadata: Box<Option<diesel_models::types::FeatureMetadata>>,
+        enable_payment_link: Option<common_enums::EnablePaymentLinkRequest>,
+        // TODO: check how to use this
+        // payment_link_config: Box<Option<api_models::admin::PaymentLinkConfigRequest>>,
+        request_incremental_authorization: Option<common_enums::RequestIncrementalAuthorization>,
+        session_expiry: Option<PrimitiveDateTime>,
+        frm_metadata: Box<Option<pii::SecretSerdeValue>>,
+        request_external_three_ds_authentication: Option<External3dsAuthenticationRequest>,
 
-#[cfg(feature = "v2")]
-#[derive(Clone, Debug, Default)]
-pub struct PaymentIntentUpdateInternal {
-    pub amount: Option<MinorUnit>,
-    pub currency: Option<storage_enums::Currency>,
-    pub status: Option<storage_enums::IntentStatus>,
-    pub amount_captured: Option<MinorUnit>,
-    pub customer_id: Option<id_type::CustomerId>,
-    pub return_url: Option<String>,
-    pub setup_future_usage: Option<storage_enums::FutureUsage>,
-    pub off_session: Option<bool>,
-    pub metadata: Option<pii::SecretSerdeValue>,
-    pub modified_at: Option<PrimitiveDateTime>,
-    pub active_attempt_id: Option<id_type::GlobalAttemptId>,
-    pub description: Option<String>,
-    pub statement_descriptor: Option<String>,
-    pub order_details: Option<Vec<pii::SecretSerdeValue>>,
-    pub attempt_count: Option<i16>,
-    pub frm_merchant_decision: Option<common_enums::MerchantDecision>,
-    pub payment_confirm_source: Option<storage_enums::PaymentSource>,
-    pub updated_by: String,
-    pub surcharge_applicable: Option<bool>,
-    pub authorization_count: Option<i32>,
-    pub session_expiry: Option<PrimitiveDateTime>,
-    pub request_external_three_ds_authentication: Option<bool>,
-    pub frm_metadata: Option<pii::SecretSerdeValue>,
-    pub customer_details: Option<Encryptable<Secret<serde_json::Value>>>,
-    pub billing_address: Option<Encryptable<Secret<serde_json::Value>>>,
-    pub shipping_address: Option<Encryptable<Secret<serde_json::Value>>>,
-    pub merchant_order_reference_id: Option<String>,
-    pub is_payment_processor_token_flow: Option<bool>,
+        // updated_by is set internally, field not present in request
+        updated_by: Box<String>,
```

Comment:
do not box types structs. It is the structs responsibility to mask pii information